.github/workflows/staging-deploy.yml

```diff
@@ -1,10 +1,15 @@
 name: Staging Deployment
 
 on:
+  pull_request:
+    branches: [staging]
+    types: [opened, reopened]
```

Comment:
I think you want to add `synchronize` to this list, to run when more changes are added to the PR.